src/TensorNetwork.jl

```diff
@@ -206,6 +224,21 @@ function neighbors(tn::TensorNetwork, i::Symbol; open::Bool=true)
     return tensors
 end
 
+const is_unsafe_region = Ref(false) # global ScopedValue for the unsafe region
```

Comment:
If we replace this `Ref` for a `ScopedValue` (from `ScopedValues.jl`), then it will also be task-safe.